bash_completion

```diff
@@ -2267,11 +2290,11 @@ _known_hosts()
 
     # NOTE: Using `_known_hosts' as a helper function and passing options
     #       to `_known_hosts' is deprecated: Use `_known_hosts_real' instead.
-    local options=""
-    [[ ${1-} == -a || ${2-} == -a ]] && options=-a
-    [[ ${1-} == -c || ${2-} == -c ]] && options+=" -c"
-    # shellcheck disable=SC2086
-    _known_hosts_real ${options-} -- "$cur"
+    local -a options=()
+    [[ ${1-} == -a || ${2-} == -a ]] && options+=(-a)
+    [[ ${1-} == -c || ${2-} == -c ]] && options+=(-c)
+    local IFS=$' \t\n' # Workaround for connected ${v+"$@"} in bash < 4.4
+    _known_hosts_real ${options[@]+"${options[@]}"} -- "$cur"
```

Comment:
I wonder what's the site where we'd need the said shellcheck disables?

Reply:
Every line. If we do not make `options` array and another function uses an array `options`, `shellcheck` will issue a warning for every line that uses `options` above.

Reply:
I can see the warnings, but what I'm wondering is what is that _another function_ that uses such an array in this context? I see `_comp_compgen_usage` and `_comp_compgen_help` declare it as an array, but neither is being invoked anywhere before `_known_hosts` that I can tell.

Or maybe is this just a naive check without considering the actual call graph, i.e. if the variable *actually* ends up ever being used as array and non-array. Maybe the "use as an array" is triggered by just the (undeclared) use as such in `_comp_compgen_help__parse` without deeper consideration.

(No objections to the change, use as an array looks cleaner to me anyway, but just want to understand what I'm looking at.)

Reply:
> Or maybe is this just a naive check without considering the actual call graph,

This is the case. That seems to be the `shellcheck` behavior.